flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/ScalarFunctionsTest.scala

```diff
@@ -1198,583 +1498,992 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
+      "ATAN2(f29, f29)",
+      math.atan2(0.46, 0.46).toString)
+
+    testAllApis(
+      atan2('f30, 'f30),
+      "atan2(f30, f30)",
+      "ATAN2(f30, f30)",
+      math.atan2(1, 1).toString)
+
+    testAllApis(
+      atan2('f31, 'f31),
+      "atan2(f31, f31)",
+      "ATAN2(f31, f31)",
+      math.atan2(-0.1231231321321321111, -0.1231231321321321111).toString)
+  }
+
   @Test
   def testDegrees(): Unit = {
-    testSqlApi(
+    testAllApis(
+      'f2.degrees(),
+      "f2.degrees()",
       "DEGREES(f2)",
       math.toDegrees(42.toByte).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f3.degrees(),
+      "f3.degrees()",
       "DEGREES(f3)",
       math.toDegrees(43.toShort).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f4.degrees(),
+      "f4.degrees()",
       "DEGREES(f4)",
       math.toDegrees(44.toLong).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f5.degrees(),
+      "f5.degrees()",
       "DEGREES(f5)",
       math.toDegrees(4.5.toFloat).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f6.degrees(),
+      "f6.degrees()",
       "DEGREES(f6)",
       math.toDegrees(4.6).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f15.degrees(),
+      "degrees(f15)",
       "DEGREES(f15)",
       math.toDegrees(-1231.1231231321321321111).toString)
   }
 
   @Test
   def testRadians(): Unit = {
-    testSqlApi(
+    testAllApis(
+      'f2.radians(),
+      "f2.radians()",
       "RADIANS(f2)",
       math.toRadians(42.toByte).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f3.radians(),
+      "f3.radians()",
       "RADIANS(f3)",
       math.toRadians(43.toShort).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f4.radians(),
+      "f4.radians()",
       "RADIANS(f4)",
       math.toRadians(44.toLong).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f5.radians(),
+      "f5.radians()",
       "RADIANS(f5)",
       math.toRadians(4.5.toFloat).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f6.radians(),
+      "f6.radians()",
       "RADIANS(f6)",
       math.toRadians(4.6).toString)
 
-    testSqlApi(
+    testAllApis(
+      'f15.radians(),
+      "radians(f15)",
       "RADIANS(f15)",
       math.toRadians(-1231.1231231321321321111).toString)
   }
 
   @Test
   def testSign(): Unit = {
-    testSqlApi(
+    testAllApis(
+      'f4.sign(),
+      "f4.sign()",
       "SIGN(f4)",
       1.toString)
 
-    testSqlApi(
+    testAllApis(
+      'f6.sign(),
+      "f6.sign()",
       "SIGN(f6)",
       1.0.toString)
 
-    testSqlApi(
+    testAllApis(
+      'f15.sign(),
+      "sign(f15)",
       "SIGN(f15)",
       "-1.0000000000000000000") // calcite: SIGN(Decimal(p,s)) => Decimal(p,s)
   }
 
   @Test
   def testRound(): Unit = {
-    testSqlApi(
+    testAllApis(
+      'f29.round('f30),
+      "f29.round(f30)",
       "ROUND(f29, f30)",
       0.5.toString)
 
-    testSqlApi(
-      "ROUND(f31, f7)",
-      "-0.123")
+    // Now not support round decimal with non-literal
```

Comment:
Is it a bug in blink or we don't support it yet? Do we need to fix this before 1.9?

Reply:
see `ReturnTypeInference.inferRound`, now we have copied calcite's type infer for round. It require round.right must be a literal...

Reply:
I don't think this is a bug